asf_tools/water_map.py

```diff
@@ -0,0 +1,100 @@
+"""Generate surface water maps from Sentinel-1 RTC products"""
+
+import argparse
+import logging
+import sys
+from pathlib import Path
+from typing import Tuple, Union
+
+import numpy as np
+from osgeo import gdal
+
+from asf_tools.composite import get_epsg_code, read_as_array, write_cog
+from asf_tools.threshold import kittler_illingworth_threshold as ki_threshold
+from asf_tools.util import tile_array
+
+log = logging.getLogger(__name__)
+
+
+def water_extent(raster: np.ndarray, tile_shape: Tuple[int, int] = (200, 200),
+                 sub_tile_shape: Tuple[int, int] = (100, 100)):
+
+    tiles = np.ma.masked_invalid(tile_array(raster, tile_shape=tile_shape, pad_value=np.nan))
+    sub_tiles_std = np.zeros((tiles.shape[0], 4))
```

Comment:
`4` should be calculated as `sub_tiles_per_tile` (`sub_tile_shape` input could be `sub_tile_count` instead?)